src/EFCore.PG.NTS/Query/ExpressionTranslators/Internal/NpgsqlNetTopologySuiteMemberTranslatorPlugin.cs

```diff
@@ -1,126 +1,125 @@
+            "Centroid"         => _sqlExpressionFactory.Function("ST_Centroid",         new[] { instance }, typeof(Point), resultGeometryTypeMapping),
+            "Count"            => _sqlExpressionFactory.Function("ST_NumGeometries",    new[] { instance }, typeof(int)),
+            "Dimension"        => _sqlExpressionFactory.Function("ST_Dimension",        new[] { instance }, typeof(Dimension)),
+            "EndPoint"         => _sqlExpressionFactory.Function("ST_EndPoint",         new[] { instance }, typeof(Point), resultGeometryTypeMapping),
+            "Envelope"         => _sqlExpressionFactory.Function("ST_Envelope",         new[] { instance }, typeof(Geometry), resultGeometryTypeMapping),
+            "ExteriorRing"     => _sqlExpressionFactory.Function("ST_ExteriorRing",     new[] { instance }, typeof(LineString), resultGeometryTypeMapping),
+            "GeometryType"     => _sqlExpressionFactory.Function("GeometryType",        new[] { instance }, typeof(string)),
+            "IsClosed"         => _sqlExpressionFactory.Function("ST_IsClosed",         new[] { instance }, typeof(bool)),
+            "IsEmpty"          => _sqlExpressionFactory.Function("ST_IsEmpty",          new[] { instance }, typeof(bool)),
+            "IsRing"           => _sqlExpressionFactory.Function("ST_IsRing",           new[] { instance }, typeof(bool)),
+            "IsSimple"         => _sqlExpressionFactory.Function("ST_IsSimple",         new[] { instance }, typeof(bool)),
+            "IsValid"          => _sqlExpressionFactory.Function("ST_IsValid",          new[] { instance }, typeof(bool)),
+            "Length"           => _sqlExpressionFactory.Function("ST_Length",           new[] { instance }, typeof(double)),
+            "NumGeometries"    => _sqlExpressionFactory.Function("ST_NumGeometries",    new[] { instance }, typeof(int)),
+            "NumInteriorRings" => _sqlExpressionFactory.Function("ST_NumInteriorRings", new[] { instance }, typeof(int)),
+            "NumPoints"        => _sqlExpressionFactory.Function("ST_NumPoints",        new[] { instance }, typeof(int)),
+            "PointOnSurface"   => _sqlExpressionFactory.Function("ST_PointOnSurface",   new[] { instance }, typeof(Geometry), resultGeometryTypeMapping),
+            "InteriorPoint"    => _sqlExpressionFactory.Function("ST_PointOnSurface",   new[] { instance }, typeof(Geometry), resultGeometryTypeMapping),
+            "SRID"             => _sqlExpressionFactory.Function("ST_SRID",             new[] { instance }, typeof(int)),
+            "StartPoint"       => _sqlExpressionFactory.Function("ST_StartPoint",       new[] { instance }, typeof(Point), resultGeometryTypeMapping),
+
+            "OgcGeometryType"  => _sqlExpressionFactory.Case(
+                _sqlExpressionFactory.Function("ST_GeometryType", new[] { instance }, typeof(string)),
+                _ogcGeometryTypeWhenThenList),
 
-            return null;
+            _ => (SqlExpression)null
```

Comment:
Or it would be better to handle all members here:

```csharp
return member.Name switch
{
    // ...
    "Count" => sqlExpressionFactory.Function(
        typeof(LineString).IsAssignableFrom(declaringType)
            ? "ST_NumPoints"
            : "ST_NumGeometries",
        new[] { instance }, typeof(int)),
    // ...

    nameof(Point.X) => Coordinate("ST_X"),
    nameof(Point.Y) => Coordinate("ST_Y"),
    nameof(Point.Z) => Coordinate("ST_Z"),
    nameof(Point.M) => Coordinate("ST_M"),

    _ => default(SqlExpression),
};

SqlExpression Coordinate(string function)
    => (typeof(Point).IsAssignableFrom(declaringType))
    ? _sqlExpressionFactory.Function(function, new[] { instance }, typeof(double))
    : null;
```

Reply:
The idea is to separate translation of properties that are only on Point vs. LineString, etc. I think it's OK like this...

Reply:
It's up to you. Just another idea how to simplify expressions.

Why do you use literals instead of `nameof`s? Is it EF style?

Reply:
> Why do you use literals instead of nameofs? Is it EF style?

Not at all - it was just this way before and I didn't notice. Changed to nameof.

This exposed that the translation tables aren't well-organized, especially in the member translator - some properties specific to a certain type of geometry are in the general switch. However, this needs to be studied a bit closer, as some properties exist on several geometry types, etc. Unfortunately I don't have time to dive into this at the moment, let's merge as-is and deal with problems as they come up - this has been working for a while without reported issues.